src/crypto/CHIPCryptoPALmbedTLS.cpp

```diff
@@ -699,8 +795,30 @@ void ClearSecretData(uint8_t * buf, size_t len)
 
 CHIP_ERROR P256Keypair::Initialize()
 {
+#if defined(MBEDTLS_USE_TINYCRYPT)
     CHIP_ERROR error = CHIP_NO_ERROR;
-    int result       = 0;
+    int result       = UECC_FAILURE;
+
+    Clear();
+
+    mbedtls_uecc_keypair * keypair = to_keypair(&mKeypair);
+    uECC_set_rng(&uecc_rng_wrapper);
```

Comment:
Why the RNG set here? This is done every time a key is used. This is odd and not super re-entrant.